### PR TITLE
Migrate `initializeQB` (`StructuredQuery` only)

### LINKED ISSUE #37391: [MLv2] [FE] Migrate initializeQB

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #37391

### Description

Adds 2 new MLv2 methods: `metricMetadata` and `segmentMetadata`. See [Slack discussion](https://metaboat.slack.com/archives/C0645JP1W81/p1704954641213749) for details.

----

There is another PR which migrates `NativeQuery`-related things in `initializeQB`: #37569.
It's in draft for now, see [Slack discussion](https://metaboat.slack.com/archives/C0645JP1W81/p1704977177544249?thread_ts=1704976478.613829&cid=C0645JP1W81).

### CODE REVIEW

File: frontend/src/metabase-lib/metrics.ts [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
 
 export function availableMetrics(query: Query): MetricMetadata[] {
   return ML.available_metrics(query);
 }
+
+export function metric(
+  query: Query,
+  metricId: MetricId,
+): MetricMetadata | null {
+  return ML.metric(query, metricId);
```
Comment (ranquild) [inline frontend/src/metabase-lib/metrics.ts:14]: `metric_metadata` to match `table_or_card_metadata`?

File: frontend/src/metabase-lib/segments.ts [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+import type { Query, SegmentMetadata } from "./types";
 
 export function availableSegments(
   query: Query,
   stageIndex: number,
 ): SegmentMetadata[] {
   return ML.available_segments(query, stageIndex);
 }
+
+export function segment(
```
Comment (ranquild) [inline frontend/src/metabase-lib/segments.ts:13]: `segment_metadata` to match `table_or_card_metadata`?

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase-lib/metrics.ts (hunk 1 above)
Comment (kamilmielnik) [reply frontend/src/metabase-lib/metrics.ts]: 8c8769650092609cc5a373d976255eb6d4ad5bce

File: frontend/src/metabase-lib/segments.ts (hunk 2 above)
Comment (kamilmielnik) [reply frontend/src/metabase-lib/segments.ts]: 8c8769650092609cc5a373d976255eb6d4ad5bce

**Review by metamben — APPROVED** [review]
BE code LGTM.

**Review by camsaul — APPROVED** [review]

**Pull request merged**